server/core/kv.go

```diff
@@ -234,6 +235,33 @@ func (kv *KV) LoadRegions(regions *RegionsInfo) error {
 	}
 }
 
+// SaveGCSafePoint saves new GC safe point to KV.
+func (kv *KV) SaveGCSafePoint(safePoint uint64) error {
+	key := path.Join(gcPath, "safe_point")
+	value := strconv.FormatUint(safePoint, 16)
+	if err := kv.Save(key, value); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+// LoadGCSafePoint loads current GC safe point from KV.
+func (kv *KV) LoadGCSafePoint() (uint64, error) {
```

Comment:
@disksing 
do we need to record the safe point in the gauge like TSO/cluster ID does?

Reply:
TiDB already has metrics for it. Just reminded that maybe we need to set up a safe value for it when we use `pd-recover` to restore PD cluster.